### PR TITLE
test: cover high-impact parser branches

### PR DESCRIPTION
**Pull request by ibgreen**
## Goals

- Make a material repository-level coverage gain by testing dense, high-impact parser and converter code.
- Keep the added CI workload fast and hermetic.
- Exercise behavior and boundary cases without removing performance-oriented branching.

## Changes

- Expand raw Arrow CSV coverage for quoted data, uneven rows, delimiter/header inference, custom quoting, late fast-path rejection, and builder growth.
- Exercise every GeoArrow union geometry kind, dimensions, null/empty collections, round trips, and selection validation.
- Cover difficult WKB triangulation inputs including self intersections, collinear vertices, narrow concavities, touching holes, and Steiner-point holes.
- Exercise GLTF scenegraph accessors, extension lifecycle, object construction, attribute mapping, and binary chunk creation.
- Cover PLY ASCII and binary scalar matrices, both endiannesses, normalized attributes, mesh lists, and invalid scalar types.
- Cover SQL relational expression, aggregate, empty-schema, scalar ordering/grouping, and join-validation branches.

## Coverage impact

Unioning the focused coverage output with the latest merged `master` CI artifact adds **285 previously uncovered statements**:

- GeoArrow geometry converter: +70
- PLY parser: +66
- GLTF scenegraph: +58
- WKB triangulation: +54
- SQL Arrow queries: +22
- CSV and supporting GIS/PLY helpers: +15

Projected statement coverage moves from **81.28% to 81.688%**, a **+0.404 percentage-point** gain.

## Validation

- `yarn lint fix`
- `yarn test-audit` — 639 files, 0 Tape, 0 violations
- Focused Chromium suite — 6 files, 99 tests passed in 3.57s
- Focused suite with repository-wide coverage instrumentation — 6 files, 99 tests passed in 7.24s
- `git diff --check`
- `yarn build` compiled all 57 module source trees, then the bundle pre-build hit the existing worktree alias issue where `ocular-bundle` resolved `@loaders.gl/crypto` against the main checkout instead of this worktree.

### CODE REVIEW

File: modules/csv/test/parse-raw-arrow-csv-bytes.spec.ts [hunk 1]
```diff
@@ -22,6 +22,80 @@ function encode(text: string): ArrayBuffer {
...
+      {name: 'beta', value: null}
+    ]);
+
+    const numericFirstRow = parseRawArrowCSVBytes(encode('1|2\n3|4'), {
+      header: 'auto',
+      dynamicTyping: true,
+      delimiter: '|',
+      columnPrefix: 'field'
+    });
+    expect(toRows(numericFirstRow!)).toEqual([{'1': '3', '2': '4'}]);
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/csv/test/parse-raw-arrow-csv-bytes.spec.ts]: Fixed in 442af168c0. The byte and ASCII fast paths now apply the same dynamic-typing auto-header classifier as the generic path, and both preserve numeric first rows under generated field names.

File: modules/gis/test/geometry-converters/wkb/convert-wkb-to-geometry.spec.ts [hunk 2]
```diff
@@ -267,6 +267,82 @@ test('triangulateWKB#rejects non-polygon geometry', async () => {
...
+      [2, 6],
+      [0, 6],
+      [0, 0]
+    ]
+  ]
+])('triangulateWKB#handles difficult %s polygons', (_name, ring) => {
+  const triangles = triangulateWKB(
+    convertGeometryToWKB({type: 'Polygon', coordinates: [ring as Position[]]})
+  );
+  expect(triangles.every(Number.isInteger)).toBe(true);
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/gis/test/geometry-converters/wkb/convert-wkb-to-geometry.spec.ts]: Fixed in 442af168c0. Valid duplicate/collinear and concave fixtures now use the existing reference earcut oracle. The intentionally self-intersecting case separately requires nonempty integer output because its duplicate closing vertex is indexed differently by the two implementations.

File: modules/ply/test/parse-ply-edge-cases.spec.ts [hunk 3]
```diff
@@ -13,6 +13,163 @@ import {
...
+      bytes.set(headerBytes);
+      bytes.set(new Uint8Array(record), headerBytes.length);
+      const table = parsePLYToArrowTable(bytes.buffer);
+
+      expect(table?.data.numRows).toBe(1);
+      expect(Array.from(table!.data.getChild('POSITION')!.get(0) as number[])).toEqual([
+        1.25, -2.5, 3.75
+      ]);
+      for (const [, name, value] of properties.slice(0, 6)) {
+        expect(table!.data.getChild(name)!.get(0)).toBeCloseTo(value, -2);
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/ply/test/parse-ply-edge-cases.spec.ts]: Fixed in 442af168c0. The UINT32 fixture now uses an exactly representable value and all integer scalar branches use exact equality.
